x-pack/test/functional/apps/rollup_job/rollup_jobs.js

```diff
@@ -4,64 +4,118 @@
+          body: {
+            '@timestamp': day.toISOString(),
+            foo_metric: 1
+          }
+        });
+      }
+    }
+
+    it('create new rollup job', async () => {
+      const indexPattern = 'to-be*';
+      const interval = '1000ms';
+
+      await log.debug('make sure all dates have the same concept of "now"');
+      await log.debug(pastDates);
+      await loadDates(pastDates);
+
       await PageObjects.common.navigateToApp('rollupJob');
+      await PageObjects.rollup.createNewRollUpJob(rollupJobName, indexPattern, indexName,
+        interval, ' ', true, { time: '*/10 * * * * ?', cron: true });
+
+      await PageObjects.common.navigateToApp('indexManagement');
+
+      await PageObjects.indexManagement.toggleRollupIndices();
+
+      const indices = await PageObjects.indexManagement.getIndexList();
+
+      await log.debug('If the index exists then the rollup job has successfully' +
+        'been created and is waiting to or has triggered.');
+
+      //If the index exists then the rollup job has successfully been created and is waiting to or has triggered.
+      const filteredIndices = indices.filter(i => i.indexName === indexName);
+      expect(filteredIndices.length).to.be.greaterThan(0);
+
+      //Stop the running rollup job.
+      await es.transport.request({
+        path: `/_rollup/job/${rollupJobName}/_stop?wait_for_completion=true`,
+        method: 'POST',
+      });
+
+
     });
 
-    after(async () => await esArchiver.unload('logstash_functional'));
+    it('create hybrid index pattern', async () => {
 
-    it('create and save a new job', async () => {
-      const jobName = 'Testjob1';
-      const indexPattern = '.kibana*';
-      const indexName = 'rollup_index';
-      const interval = '1000ms';
+      await log.debug('Create data for the rollup job to recognize.');
+
+      await pastDates.map(async (day) => {
+        await es.index(await mockRolledUpData(rollupJobName, day));
+      });
+      await PageObjects.common.navigateToApp('rollupJob');
+
+      //Index live data to be used in the test.
+
+      await log.debug('Add data for 1,2 and 3 days into the future.');
+      const futureDates = [
+        datemath.parse('now', { forceNow: now }),
+        datemath.parse('now+1d', { forceNow: now }),
+        datemath.parse('now+2d', { forceNow: now }),
+        datemath.parse('now+3d', { forceNow: now }),
+      ];
+
+      await loadDates(futureDates, 'live-data');
+
+      await PageObjects.common.navigateToApp('settings');
+      await PageObjects.settings.createIndexPattern('live*,' + indexName, '@timestamp', false);
+
+      await PageObjects.common.navigateToApp('discover');
+      const hits = await PageObjects.discover.getHitCount();
+
+      //Number of hits should be at least 7. Due to CI being slow, the legacy data may not be there.
+      //3 for rolled up data, 3 for legacy data that was rolled up, and 4 for the live data.
+      expect(Number.parseInt(hits)).to.be.above(6);
```

Comment:
Is it a good enough check? It basically checks the number of documents and does not check if its data is correct. Don't we want to assure that rolled up data is in place and can we do it?  /cc @LeeDr @rasroh

Reply:
The test is just making sure that we can create an index pattern that combines a rollup and live data. There are going to be other tests from your team that will provide checks of data integrity and compatibility for visualizations. Lee and I went over this and figured that this was a sufficient check.